usr/src/cmd/getent/dogetnetgr.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * This file and its contents are supplied under the terms of the
+ * Common Development and Distribution License ("CDDL"), version 1.0.
+ * You may only use this file in accordance with the terms of version
+ * 1.0 of the CDDL.
+ *
+ * A full copy of the text of the CDDL should have accompanied this
+ * source.  A copy of the CDDL is also available via the Internet at
+ * http://www.illumos.org/license/CDDL.
+ */
+
+/*
+ * Copyright 2012 Nexenta Systems, Inc.  All rights reserved.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <netdb.h>
+#include "getent.h"
+
+int
+dogetnetgr(const char **list)
+{
+	char *host;
+	char *user;
+	char *dom;
+	int rc = EXC_SUCCESS;
+
+	if (list == NULL || *list == NULL)
+		return (EXC_ENUM_NOT_SUPPORTED);
+
+	for (; *list != NULL; list++) {
+		if (setnetgrent(*list) != 0)
+			return (EXC_ENUM_NOT_SUPPORTED);
+		printf("%s", *list);
+                while (getnetgrent(&host, &user, &dom) != 0) {
```

Comment:
Something has gone wrong with spacing here (tab followed by spaces)
pbchk will probably catch it anyway.